exercises/practice/circular-buffer/tests/Tests.elm

```diff
@@ -0,0 +1,159 @@
+                \() ->
+                    run 3 [ Write 1, Write 2, Read, Write 3, Read, Read ]
+                        |> Expect.equal
+                            [ WriteSuccess
+                            , WriteSuccess
+                            , ReadSuccess 1
+                            , WriteSuccess
+                            , ReadSuccess 2
+                            , ReadSuccess 3
+                            ]
+        , skip <|
+            test "items cleared out of buffer can't be read" <|
+                \() ->
+                    run 1 [ Write 1, Clear, Read ]
+                        |> Expect.equal [ WriteSuccess, ReadFailure ]
+        , skip <|
+            test "clear frees up capacity for another write" <|
+                \() ->
+                    run 1 [ Write 1, Clear, Write 2, Read ]
+                        |> Expect.equal [ WriteSuccess, WriteSuccess, ReadSuccess 2 ]
+        , skip <|
+            test "clear does nothing on empty buffer" <|
+                \() ->
+                    run 1 [ Clear, Write 1, Read ]
+                        |> Expect.equal [ WriteSuccess, ReadSuccess 1 ]
+        , skip <|
+            test "overwrite acts like write on non-full buffer" <|
+                \() ->
+                    run 2 [ Write 1, Overwrite 2, Read, Read ]
+                        |> Expect.equal [ WriteSuccess, ReadSuccess 1, ReadSuccess 2 ]
+        , skip <|
+            test "overwrite replaces the oldest item on full buffer" <|
+                \() ->
+                    run 2 [ Write 1, Write 2, Overwrite 3, Read, Read ]
+                        |> Expect.equal [ WriteSuccess, WriteSuccess, ReadSuccess 2, ReadSuccess 3 ]
+        , skip <|
+            test "overwrite replaces the oldest item remaining in buffer following a read" <|
+                \() ->
+                    run 3 [ Write 1, Write 2, Write 3, Read, Write 4, Overwrite 5, Read, Read, Read ]
+                        |> Expect.equal
+                            [ WriteSuccess
+                            , WriteSuccess
+                            , WriteSuccess
+                            , ReadSuccess 1
+                            , WriteSuccess
+                            , ReadSuccess 3
+                            , ReadSuccess 4
+                            , ReadSuccess 5
+                            ]
+        , skip <|
+            test "initial clear does not affect wrapping around" <|
+                \() ->
+                    run 2 [ Clear, Write 1, Write 2, Overwrite 3, Overwrite 4, Read, Read, Read ]
+                        |> Expect.equal
+                            [ WriteSuccess
+                            , WriteSuccess
+                            , ReadSuccess 3
+                            , ReadSuccess 4
+                            , ReadFailure
+                            ]
+        , skip <|
+            test "buffer can store strings" <|
```

Comment:
This test doesn't exist in the canonical data, but one of the comments encourages tracks to add it for languages that can handle polymorphism. Seemed like worth doing.